Entity-specific anonymization

maximilianmozes opened the pull request:
I added functionality to specify the entities to be included in the anonymization for three anonymization types (context-preserving, named entity-based and combined). README and example (run.js) are adjusted accordingly. Moreover, I added a Logging class that might be helpful for further extensions. Please **review** and **test** carefully. Thanks!

Commit:
added entity-specific anonymization and logging class

Commit:
readme changes, version update and function renaming

Commit:
readme changes

Commit:
minifix README

Commit:
Updated example

ben-aaron188 commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "netanos",
-  "version": "1.1.5",
+  "version": "1.1.3",
```
@maximilianmozes This is not on par with the npm version. The next one will then be 1.1.6.

ben-aaron188 reviewed (changes requested):
@maximilianmozes Fantastic. A few things I noticed:
- the `entities` object (or limitations in the interface code) is really helpful but it brings some disambiguation issues:
- - if we disable PERSON, "Ben" is identified as OTHER (so maybe add OTHER in the `entities` object too?)
- - if we disable DATE, "1000 hours" is identified as NUMERIC

If suspect there are other issues, and I'd suggest adding the other cases (numerics, pronouns, other, etc.] to the `entities` object as booleans too.

Commit:
added other, numeric and pronouns to entity specifications

Commit:
mini-fixes

maximilianmozes commented:
@ben-aaron188 thanks for the review. I have added the remaining cases to the entities object and adjusted the version.

ben-aaron188 reviewed (changes requested):
Really good. One minor conflict between the entity-specifications (referring to the test string):
- if we set `person: false`, the second name "Ben" is passed on to the `OTHER` entity group. I suspect that this is because it's a leftover with a capital letter.
- A solution could be this: if person/locations/[all that can in principle start with a capital letter - but we could limit it to `PERSON`, `ORGANIZATION`, and `LOCATION`] is set to `false`, we exclude any `PERSON`, `ORGANIZATION`, and `LOCATION` entities from the `OTHER` group.
- - then "Ben" would be excluded from any `OTHER` anonymization.

What do you think?

Commit:
extended entity specifications

maximilianmozes commented:
@ben-aaron188 et voilà. I adjusted the code accordingly. Please have a look and let me know what you think. Thanks!

ben-aaron188 reviewed (changes requested):
I think a minor conflict remains:

Check in `run.js` with these parameters in the entities object:

    var entities = {
      person: true,
      organization: true,
      currency: true,
      date: true,
      location: false,
      pronoun: true,
      numeric: true,
      other: true
    };

Commit:
punctuation fix

Commit:
extended example

maximilianmozes commented:
Thanks @ben-aaron188! I did not cover the case that an entity contains a dot (in the example the input string contains the word "Amsterdam." instead of "Amsterdam" and hence the replacement did not work). Fixed this now. I also fixed another minor issue.
